domain/domain.go

```diff
@@ -256,20 +272,63 @@ func (c *ddlCallback) OnChanged(err error) error {
 	}
 	log.Warnf("[ddl] on DDL change")
 
-	c.do.mustReload()
-	return nil
+	return c.do.MustReload()
+}
+
+type schemaValidityInfo struct {
+	isValid          bool
+	lastInvalidTS    uint64
```

Comment:
If one txn begins before server loads schema failed and it finishes after server reloads schema successfully, then need lastInvalidTS to make sure this txn is invalid.